tools/cli-generator/generate.go

```diff
@@ -0,0 +1,281 @@
+	}
+
+	return filepath.Join(basePath, "internal", "cli", filepath.Join(c.CommandPaths()...))
+}
+
+func (c *Command) FileName(basePath string) string {
+	return c.baseFileName(basePath) + ".go"
+}
+
+func (c *Command) UnitTestFileName(basePath string) string {
+	return c.baseFileName(basePath) + "_test.go"
+}
+
+func fileExists(f string) bool {
+	_, err := os.Stat(f)
+	return !os.IsNotExist(err)
+}
+
+func (cli *CLI) generateStore(store *Store) error {
+	storeFile := filepath.Join(cli.basePath, "internal", "store", store.BaseFileName+".go")
+
+	fileCreated, err := cli.generateFile(storeFile, store.TemplateFile(), store)
+	if err != nil {
+		return err
+	}
+	if !fileCreated {
+		return nil
+	}
+	return goGenerate(storeFile)
+}
+
+func (cli *CLI) generateFile(file string, templateFile string, data any) (bool, error) {
+	if !cli.overwrite && fileExists(file) {
+		fmt.Printf("File %q already present in disk, skipping\n", file)
+		return false, nil
+	}
+
+	err := os.MkdirAll(filepath.Dir(file), os.ModePerm)
+	if err != nil {
+		return false, err
+	}
+
+	f, err := os.Create(file)
+	if err != nil {
+		return false, err
+	}
+
+	defer f.Close()
+
+	err = cli.templates.ExecuteTemplate(f, templateFile, data)
+	if err != nil {
+		return false, err
+	}
+
+	err = cleanupFile(file)
+	if err != nil {
+		return true, err
+	}
+
+	return true, nil
+}
+
+func (cli *CLI) generateCommand(cmd *Command) error {
+	_, err := cli.generateFile(cmd.FileName(cli.basePath), cmd.TemplateFile(), cmd)
+	if err != nil {
+		return err
+	}
+
+	_, err = cli.generateFile(cmd.UnitTestFileName(cli.basePath), cmd.TemplateUnitTestFile(), cmd)
+	if err != nil {
+		return err
+	}
+
+	for i := range cmd.SubCommands {
+		if err := cli.generateCommand(&cmd.SubCommands[i]); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func cleanupFile(filePath string) error {
+	execCmd := exec.Command("goimports", "-w", filePath)
+	_, err := execCmd.Output()
+	if err != nil {
+		return err
+	}
+
+	execCmd = exec.Command("gofmt", "-w", "-s", filePath)
+	_, err = execCmd.Output()
+	return err
+}
+
+func goGenerate(filePath string) error {
+	execCmd := exec.Command("go", "generate", filePath)
+	_, err := execCmd.Output()
+	return err
+}
+
+func runMake() error {
+	execCmd := exec.Command("make", "gen-docs")
+	_, err := execCmd.Output()
+	return err
+}
+
+func newCli(overwrite bool) (*CLI, error) {
+	cli := CLI{overwrite: overwrite}
+	err := yaml.Unmarshal(spec, &cli)
+	if err != nil {
+		return nil, err
+	}
+	cli.basePath, err = filepath.Abs("./")
+	if err != nil {
+		return nil, err
+	}
+
+	cli.templates, err = template.New("root").Funcs(templateFuncs).ParseFS(templateFolder, "templates/*.txt")
+	if err != nil {
+		return nil, err
+	}
+
+	return &cli, nil
+}
+
+func (cli *CLI) generateCli() error {
+	for i := range cli.Stores {
+		if err := cli.generateStore(&cli.Stores[i]); err != nil {
+			return fmt.Errorf("%w: %s", ErrGenerateCli, err)
```

Comment:
FYI intellij complains about this when we move to go1.20 we could do 
```suggestion
			return fmt.Errorf("%w: %w", ErrGenerateCli, err)
```

but I think we should stick to 
```suggestion
			return fmt.Errorf("error generating store: %w", ErrGenerateCli, err)
```

Reply:
using this I get the error `Error: tools/cli-generator/generate.go:273:11: fmt.Errorf call has more than one error-wrapping directive %w` on github actions

Reply:
I guess we are still on `go 1.19`

Reply:
and hence the 
>when we move to go1.20 we could do